### ISSUE TITLE
Android native crash in __android_log_print when using our own compiled mmkv.so

### ISSUE THREAD

**Issue by samlu** [Tencent/MMKV#320]
I experienced the same issue as the following ticket:
https://github.com/Tencent/MMKV/issues/239
 
MMKV source version: 1.0.22
NDK: 20.0
build .so on Windows by using "gradlew build"

The crash is caused by the following line in MMKVLog.h

`#define __filename__ (strrchr(__FILE__, '/') + 1)`

It should be changed as

```
#ifdef __UNIX__
#define __filename__ (strrchr(__FILE__, '/') + 1)
#else 
#define __filename__ (strrchr(__FILE__, '\\') + 1)
#endif 
```

Please also check if there are similar platform issues.
 
```
A/DEBUG: *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
A/DEBUG: Build fingerprint: 'google/hammerhead/hammerhead:6.0.1/xxx/release-keys'
A/DEBUG: Revision: '11'
A/DEBUG: ABI: 'arm'
A/DEBUG: pid: 20764, tid: 20764, name: xxx >>> xxx <<<
A/DEBUG: signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x1
A/DEBUG:     r0 00000001  r1 00000001  r2 00000000  r3 00000007
A/DEBUG:     r4 00000002  r5 bed7f6a0  r6 0ccccccc  r7 ffffffff
A/DEBUG:     r8 bed7ffdc  r9 00000000  sl 00000001  fp bed7ffe0
A/DEBUG:     ip 80000000  sp bed7f598  lr b6cc2775  pc b6ca087a  cpsr a00b0030
A/DEBUG: backtrace:
A/DEBUG:     #00 pc 0001887a  /system/lib/libc.so (strlen+21)
A/DEBUG:     #01 pc 0003a771  /system/lib/libc.so (__vfprintf+2948)
A/DEBUG:     #02 pc 0003db47  /system/lib/libc.so (vsnprintf+122)
A/DEBUG:     #03 pc 00002c2f  /system/lib/liblog.so (__android_log_print+42)
A/DEBUG:     #04 pc 00026761  /data/app/xxx/lib/arm/libmmkv.so (_Z17_MMKVLogWithLevel12MMKVLogLevelPKcS1_iS1_z+360)
A/DEBUG:     #05 pc 0001715b  /data/app/xxx/lib/arm/libmmkv.so (JNI_OnLoad+842)
A/DEBUG:     #06 pc 0025b387  /system/lib/libart.so (_ZN3art9JavaVMExt17LoadNativeLibraryEP7_JNIEnvRKNSt3__112basic_stringIcNS3_11char_traitsIcEENS3_9allocatorIcEEEEP8_jobjectPS9_+1238)
A/DEBUG:     #07 pc 002d13d7  /system/lib/libart.so (_ZN3artL18Runtime_nativeLoadEP7_JNIEnvP7_jclassP8_jstringP8_jobjectS5_+194)
A/DEBUG:     #08 pc 73068105  /data/dalvik-cache/arm/system@framework@boot.oat (offset 0x1ed6000)
```

**Comment by lingol** [comment]
Already fixed in dev.

**Comment by samlu** [comment]
Thank you for the update. It works.

**Comment by lingol** [comment]
Released with [v1.0.23](https://github.com/Tencent/MMKV/releases/tag/v1.0.23).
